Mathlib/Topology/Algebra/Order/Group.lean

```diff
@@ -7,120 +7,131 @@ import Mathlib.Topology.Order.LeftRightNhds
-  fun x hx => (h x hx).abs
-
-theorem tendsto_abs_nhdsWithin_zero : Tendsto (abs : G → G) (𝓝[≠] 0) (𝓝[>] 0) :=
-  (continuous_abs.tendsto' (0 : G) 0 abs_zero).inf <|
-    tendsto_principal_principal.2 fun _x => abs_pos.2
+        |c * d / (a * b)|ₘ = |(c / a) * (d / b)|ₘ := by rw [div_mul_div_comm]
+        _ ≤ |c / a|ₘ * |d / b|ₘ := mabs_mul ..
+        _ < δ * (ε / δ) := mul_lt_mul_of_lt_of_lt hc hd
+        _ = ε := mul_div_cancel ..
+    · have (x : G) : ∀ᶠ y in 𝓝 x, y = x :=
+        (eventually_mabs_div_lt _ hε).mono fun y hy ↦ mabs_div_le_one.mp <| hε_min _ hy
+      filter_upwards [(this _).prod_nhds (this _)]
+      simp [hε]
+  continuous_inv := continuous_iff_continuousAt.2 fun a ↦
+    LinearOrderedCommGroup.tendsto_nhds.2 fun ε ε0 ↦
+      (eventually_mabs_div_lt a ε0).mono fun x hx ↦ by rwa [inv_div_inv, mabs_div_comm]
+
+@[to_additive (attr := continuity)]
+theorem continuous_mabs : Continuous (mabs : G → G) :=
+  continuous_id.max continuous_inv
+
+section Tendsto
+
+variable {α : Type*} {l : Filter α} {f : α → G}
+
+@[to_additive]
+protected theorem Filter.Tendsto.mabs {a : G} (h : Tendsto f l (𝓝 a)) :
+    Tendsto (fun x => |f x|ₘ) l (𝓝 |a|ₘ) :=
+  (continuous_mabs.tendsto _).comp h
+
+@[to_additive]
+theorem tendsto_one_iff_mabs_tendsto_one (f : α → G) :
+    Tendsto f l (𝓝 1) ↔ Tendsto (mabs ∘ f) l (𝓝 1) := by
+  refine ⟨fun h => (mabs_one : |(1 : G)|ₘ = 1) ▸ h.mabs, fun h => ?_⟩
+  have : Tendsto (fun a => |f a|ₘ⁻¹) l (𝓝 1) := (inv_one : (1 : G)⁻¹ = 1) ▸ h.inv
+  exact tendsto_of_tendsto_of_tendsto_of_le_of_le this h (fun x => inv_mabs_le <| f x) fun x =>
+    le_mabs_self <| f x
+
+end Tendsto
+
+variable {X : Type*} [TopologicalSpace X] {f : X → G} {s : Set X} {x : X}
+
+@[to_additive (attr := fun_prop)]
+protected theorem Continuous.mabs (h : Continuous f) : Continuous fun x => |f x|ₘ :=
+  continuous_mabs.comp h
+
+@[to_additive (attr := fun_prop)]
+protected theorem ContinuousAt.mabs (h : ContinuousAt f x) : ContinuousAt (fun x => |f x|ₘ) x :=
+  Filter.Tendsto.mabs h
+
+@[to_additive]
+protected theorem ContinuousWithinAt.mabs (h : ContinuousWithinAt f s x) :
+    ContinuousWithinAt (fun x => |f x|ₘ) s x :=
+  Filter.Tendsto.mabs h
+
+@[to_additive (attr := fun_prop)]
+protected theorem ContinuousOn.mabs (h : ContinuousOn f s) : ContinuousOn (fun x => |f x|ₘ) s :=
+  fun x hx => (h x hx).mabs
+
+@[to_additive]
+theorem tendsto_mabs_nhdsNE_one : Tendsto (mabs : G → G) (𝓝[≠] 1) (𝓝[>] 1) :=
+  (continuous_mabs.tendsto' (1 : G) 1 mabs_one).inf <|
+    tendsto_principal_principal.2 fun _x => one_lt_mabs.2
+
+@[deprecated (since := "2025-03-18")]
+alias tendsto_abs_nhdsWithin_zero := tendsto_abs_nhdsNE_zero
 
 /-- In a linearly ordered additive group, the integer multiples of an element are dense
 iff they are the whole group. -/
-theorem denseRange_zsmul_iff_surjective {a : G} :
-    DenseRange (· • a : ℤ → G) ↔ Surjective (· • a : ℤ → G) := by
+@[to_additive]
```

Comment:
Shouldn't this docstring be moved into `to_additive` and a new one added about integer powers of an element? (Same for `not_denseRange_zpow` below)